tapgarden/caretaker.go

```diff
@@ -1150,6 +1152,11 @@ func (b *BatchCaretaker) storeMintingProof(ctx context.Context,
 		ScriptKey: &a.ScriptKey,
 	}
 
+	var proofBuf bytes.Buffer
+	if err = mintingProof.Encode(&proofBuf); err != nil {
```

Comment:
I think we have some lingering double encoding here and in the planter - on L1102 above we encode the same proof as a file with `EncodeAsProofFile`, which calls `Enccode` with the proof internally.

If we reworked `NewFile()` and `EncodeAsProofFile()` to accept already-encoded proofs we could save on one `Encode()` call. Can add to a follow-up PR.

Reply:
Ah yeah I think that can indeed return the main proof there. I think ultimately it's worth it more to optimize the proof fetching vs minting here. But we can do another path to basically re-use the serialization more.